cmd/podman/kube/play.go

```diff
@@ -265,7 +271,33 @@ func play(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	return kubeplay(reader)
+	if err = kubeplay(reader); err != nil {
+		return err
+	}
+
+	if playOptions.Wait {
```

Comment:
I find "wait" confusing.  It implies to me that Podman will actually wait for the pods to stop and then clean up; or clean up on receiving SIGTERM.  I didn't test it but it looks like nothing will happen when all pods have stopped.

Reply:
Maybe it should imply `--service-container` and wait for it to stop/exit.  This way, Podman returns when all pods are done without having to press CTRL+C.

Reply:
If I just ran with --service-container now, is that enough to implement the --wait functionality?  IE would ^C shut down the containers?

Reply:
No, we need to wait for the service container to finish. That means we need two goroutines: one waiting for the container, another listening on the signal handler. If one kicks in, the other should be terminated as well.

The name of the service container is predictable (see https://github.com/containers/podman/blob/main/pkg/domain/infra/abi/play.go#L105), so we know what to wait for.

Reply:
If wait fails with "no such container" _after_ running kube-play, we can safely assume the service container has exited (and was removed).

Reply:
It doesn't clean up when the pods have stopped, the plan is to add that in a follow up PR and have this PR only focus on --wait. 
The last PR that was opened for this was using a --detach flag but after discussions on it consensus was to use --wait https://github.com/containers/podman/pull/15089.

Reply:
I will fix the description for --wait to say it only cleans up when a sigterm is received.

Reply:
I don't find it useful as in its current form --wait would run indefinitely and only return when receiving SIGTERM. I don't think that's a nice user experience. Probably I'm missing the point.

Reply:
From the issue, looks like this is how docker-compose behaves too and that was the ask for kube play. Cleaning up after the containers exit is the next step - it would be a bigger change so that is why this adds the cleanup on interrupt first and then we can continue to add cleanup when containers have exited.

Reply:
I cannot extract this behavior from the issue.  I read the request to stop the workload on SIGTERM but I don't see that Podman should continue running even when the workload has already finished.

In the end, it really just boils down to waiting for the "service container" to finish.  Plus some minor tweaks in the kube backend since the --service-container currently assumes to be running inside of a systemd unit.

Reply:
I agree, I think --wait should wait until the job completes, or exit when the user hit's ^c.

I don't see a reason to keep the podman process running when all of its children are complete.

Reply:
Fixed